AGW-4339: Release WSDL Proxy 2.1.0 (#277)
mulesoft/docs-api-manager: modules/ROOT/pages/building-soap-proxy.adoc

</types>
----

> 

Reviewer: So...does the intro text make clear that there are two METHODS of retrieving WSDL: either by using user credentials or by using TLS?

Reviewer 2: Done! Thanks!

> == Retrieving a WSDL Using Basic Authentication Credentials

Reviewer: ```suggestion
== Retrieving a WSDL by Using Basic Authentication Credentials
```

Reviewer 2: Done! Thanks!

> You can retrieve a WSDL, hosted on a remote site, protected with a basic authentication by using the SOAP proxy mechanism. You configure the `basic.auth.user` and `basic.auth.password` application properties to retrieve the WSDL. These properties are set in the `basicauth.properties` config file and are passed to the validator module:

Reviewer: ```suggestion
You can retrieve a protected WSDL that is hosted on a remote site by using the SOAP proxy mechanism to configure the `basic.auth.user` and `basic.auth.password` application properties in the `basicauth.properties` configuration file. The properties are then passed to the validator module:
```

Reviewer 2: Done! Thanks!

[source,xml,linenums]
----
<soap-request-validator:config name="validator-config" wsdlLocation="http://myremoteurl:9092/service?wsdl" serviceName="myServiceName"
                                   servicePort="myServicePort" schemaValidationEnabled="true"
                                   basicAuthUser="${basic.auth.user}" basicAuthPassword="${basic.auth.password}"/>
----

> If no authentication values are provided for the properties, the WSDL is retrieved without the basic authentication.

Reviewer: Does "without the basic authentication" mean that the WSDL is not authenticated but still retrieved? 
```suggestion
If you do not provide authentication values for the properties, the WSDL is retrieved unauthenticated.
```
What does that mean? 
This topic is about retrieving the WSDL by using authentication, so why are we mentioning NOT providing properties and NOT using authentication?
And how does that work, to retrieve the WSDL unauthenticated?

Reviewer 2: Done! Thanks!

</types>
----

== Retrieving a WSDL using Basic Authentication credentials

The SOAP proxy provides a mechanism for retrieving a WSDL hosted on a remote site protected with Basic Auth. This is achieved by configuring the application properties `basic.auth.user` and `basic.auth.password`.

These properties are set in the `basicauth.properties` config file and passed to the validator module.

[source,xml,linenums]
----
<soap-request-validator:config name="validator-config" wsdlLocation="http://myremoteurl:9092/service?wsdl" serviceName="myServiceName"
                                   servicePort="myServicePort" schemaValidationEnabled="true"
                                   basicAuthUser="${basic.auth.user}" basicAuthPassword="${basic.auth.password}"/>
----

If no value is provided for the properties, then the WSDL retrieval is done without Basic Authentication.

== Retrieving a WSDL using TLS

> As from version 2.1.0, the SOAP Proxy supports sas an out of the box feature retrieving a WSDL from a remote location reusing the TLS context configuration set in API Manager.

Reviewer 3: emphasize that it will be reusing the TLS Context of the proxy plis.

</types>
----

== Retrieving a WSDL using Basic Authentication credentials

The SOAP proxy provides a mechanism for retrieving a WSDL hosted on a remote site protected with Basic Auth. This is achieved by configuring the application properties `basic.auth.user` and `basic.auth.password`.

These properties are set in the `basicauth.properties` config file and passed to the validator module.

[source,xml,linenums]
----
<soap-request-validator:config name="validator-config" wsdlLocation="http://myremoteurl:9092/service?wsdl" serviceName="myServiceName"
                                   servicePort="myServicePort" schemaValidationEnabled="true"
                                   basicAuthUser="${basic.auth.user}" basicAuthPassword="${basic.auth.password}"/>
----

If no value is provided for the properties, then the WSDL retrieval is done without Basic Authentication.

== Retrieving a WSDL using TLS

> In order to retrieve a WSDL using a TLS context, then the validator module present in the proxy must be configured using the `requesterConfig` parameter.

Reviewer 3: This section is exposing internal implementation that the customer does not care. We should just state that starting from version blabla we support OOTB the TLS Scenario by reusing configuration set in APIM

                                   basicAuthUser="${basic.auth.user}" basicAuthPassword="${basic.auth.password}"/>
----

If no authentication values are provided for the properties, the WSDL is retrieved without the basic authentication.

== Retrieving a WSDL Using TLS

> With version 2.1.0, the SOAP proxy is now provided as an out-of-the-box feature to retrieve a WSDL from a remote location by reusing the TLS context configuration that is set in API Manager. The TLS context configured for the outbound configuration of the proxy is used for the operation.

Author: Dazzling! This is so much better than what I had originally written. Thank you very much, Priya!

Reviewer: With version 2.1.0 of what? Best guess...:
```suggestion
<what?> version 2.1.0 and later includes the SOAP proxy necessary to retrieve a WSDL from a remote location. Instead of you configuring authentication, the TLS context configured for the outbound configuration of the proxy is used to retrieve the WSDL.
```
In <what?> version 2.1.0 and later, you no longer need to configure basic authentication

Reviewer 2: Done! Thanks!
